internal/generator/backend/template/zkpschemes/groth16/groth16.marshal.go.tmpl

```diff
@@ -178,35 +180,39 @@ func (vk *VerifyingKey) readFrom(r io.Reader, raw bool) (int64, error) {
 		&nbCommitments,
 	}
 
-	for _, v := range toDecode {
+	for i, v := range toDecode {
 		if err := dec.Decode(v); err != nil {
-			return dec.BytesRead(), err
+			return dec.BytesRead(), fmt.Errorf("read field %d: %w", i, err)
 		}
 	}
 
 	vk.PublicAndCommitmentCommitted = utils.Uint64SliceSliceToIntSliceSlice(publicCommitted)
 	
-	vk.CommitmentKeys = make([]pedersen.VerifyingKey, nbCommitments)
```

Comment:
what's the "typical" nbCommitments value? 
for perf, we could add
```
if nbCommitments < N {
vk.CommitmentKeys = make(..., 0, nbCommitments)
}
```

Reply:
It is usually very little a la 0 or 1. So I assumed that doing `append` would be OK as the slices are allocated with bigger cap usually anyway.

There was actually another issue - we failed the roundtrip tests when we had `vk.CommitmentKeys == nil` or `vk.CommitmentKeys = []pedersen.VerificationKey`. By default when we initialize the Groth16 key in the roundtrip tests we only do `new(VerificationKey)` and get failing tests. We could have a separate case for `nbCommitments == 0`, but dunno if it is worth it.